packages/client/src/index.ts

```diff
@@ -69,46 +69,52 @@ export default class LogtoClient {
     redirectUri: string,
     onRedirect: (url: string) => void = createDefaultOnRedirect()
   ) {
-    const { url, codeVerifier } = await getLoginUrlAndCodeVerifier({
+    const { url, codeVerifier, state } = await getLoginUrlAndCodeVerifier({
       baseUrl: this.oidcConfiguration.authorization_endpoint,
       clientId: this.clientId,
       scope: this.scope,
       redirectUri,
     });
-    this.sessionManager.set({ redirectUri, codeVerifier });
+    this.sessionManager.set({ redirectUri, codeVerifier, state });
     onRedirect(url);
   }
 
   public isLoginRedirect(url: string) {
     try {
-      const { code, error } = parseRedirectCallback(url);
+      const { code, state, error } = parseRedirectCallback(url);
+      if (!code || !state || error) {
+        return false;
+      }
 
-      if (error || !code) {
+      const session = this.sessionManager.get();
+      if (!session) {
         return false;
       }
-    } catch {
-      return false;
-    }
 
-    const session = this.sessionManager.get();
+      const { redirectUri, state: originalState } = session;
+      if (state !== originalState) {
+        return false;
+      }
 
-    if (!session) {
+      return url.startsWith(redirectUri);
+    } catch {
       return false;
     }
-
-    const { redirectUri } = session;
-    return url.startsWith(redirectUri);
   }
 
   public async handleCallback(url: string) {
-    const { code, error, error_description } = parseRedirectCallback(url);
+    const { code, state, error, error_description } = parseRedirectCallback(url);
 
     if (error) {
       throw new Error(error_description ?? error);
     }
 
     if (!code) {
-      throw new Error(`Can not found authorization_code in url: ${url}`);
+      throw new Error(`Authorization_code not found in url: ${url}`);
```

Comment:
🤣  is this change just error message text?

Reply:
Yes.